testing/util/altair.go

```diff
@@ -392,22 +398,17 @@ func GenerateFullBlockAltair(
 		slot = currentSlot + 1
 	}
 
-	syncAgg, err := generateSyncAggregate(bState, privs, parentRoot)
+	stCopy := bState.Copy()
+	stCopy, err = transition.ProcessSlots(context.Background(), stCopy, slot)
```

Comment:
These functions would increase the slot  temporarily to obtain the block proposer. While this would mostly work, it fails in producing the right signatures at fork transition and most probably epoch boundaries as well. Full process slots are needed. These functions became slower because of this and the extra needed copy.